tests/tracer/test_tracer.py

```diff
@@ -1702,3 +1703,21 @@ def process_trace(self, trace):
     # Force gc
     gc.collect()
     assert len(spans) == 0
+
+
+def test_top_level(tracer):
```

Comment:
We already have a few tests which check for top level. It might make sense to refactor/combine these tests: https://github.com/DataDog/dd-trace-py/blob/1.x/tests/tracer/test_processors.py#L228. Just an idea. I don't think this should block this PR

Reply:
yeah I'm ok with the slight duplication as the existing tests test the integration of the logic into the processor.